stdlib/Random/src/Xoshiro.jl

```diff
@@ -122,6 +122,7 @@ rng_native_52(::TaskLocalRNG) = UInt64
 copy(rng::Union{TaskLocalRNG, Xoshiro}) = Xoshiro(getstate(rng)...)
 copy!(dst::Union{TaskLocalRNG, Xoshiro}, src::Union{TaskLocalRNG, Xoshiro}) = setstate!(dst, getstate(src))
 ==(x::Union{TaskLocalRNG, Xoshiro}, y::Union{TaskLocalRNG, Xoshiro}) = getstate(x) == getstate(y)
+hash(x::Union{TaskLocalRNG, Xoshiro}, h::UInt) = hash(getstate(x), h + 0x49a62c2dda6fa9be % UInt)
```

Comment:
What's the magic number for? If it's required, please document why it's there.

Reply:
It's essentially so that `hash(xoshiro) != hash(getstate(xoshiro))`, i.e. the hash depends not only on the raw contained data, but also on the "meaning" of the data, i.e. the type. At least I believe that's the reason why we do this kind of things for a bunch of base types, look at `Base.hasha_seed` (abstractdict.jl), `hashs_seed` (set.jl), `hash_abstractarray_seed`, `hashre_seed`. But it's not required per se.

Reply:
None of those have a comment for why or how those constants are chosen :pensive:  At the very least, those constructions actively choose a different value when on non-64-bit Julia; it's not just `% UInt`:

```
julia> const hashre_seed = UInt === UInt64 ? 0x67e195eb8555e72d : 0xe32373e4
0x67e195eb8555e72d

julia> hashre_seed % UInt32
0x8555e72d
```

(from the regex hashing.)

Reasoning is good, but should be documented with a comment to reduce the number of undocumented magic numbers in Base.

Reply:
Yes I don't understand why these choose a totally different value on 32 vs 64 bits platform, as opposed to what I did here (`magic % UInt`). OK I can leave a terse comment here.

Reply:
one reason to chose a constant is that you want to make sure the constant doesn't have small prime factors due to superstition.

Reply:
"a" constant doesn't provide that though - [nothing up my sleeve numbers](https://en.m.wikipedia.org/wiki/Nothing-up-my-sleeve_number) are certainly not randomly chosen, and if that's the intent, these numbers fail that principle (and should be even better documented).